### PR TITLE
feat: creating new function that will be called by other functions to allow for data aggregation

### PR DESCRIPTION
**Pull request by SamRobinson75684**
… for the aggregation of data

## Description

creating new function that will be called by other functions to allow for the aggregation of data

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [x] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/src/Functions/screeningDataServices/GetValidationExceptions/GetValidationExceptions.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public HttpResponseData Run([HttpTrigger(AuthorizationLevel.Anonymous, "get", "p
     {
         foreach (var exception in _validationData.GetAll())
         {
-            _logger.LogInformation($"Exception {exception.RuleName} at {exception.DateCreated}");
```
Comment (stephhou) [inline application/CohortManager/src/Functions/screeningDataServices/GetValidationExceptions/GetValidationExceptions.cs:28]: Just wondering why this file has been changed.

Comment (SamRobinson75684) [reply application/CohortManager/src/Functions/screeningDataServices/GetValidationExceptions/GetValidationExceptions.cs:28]: this was causing build errors as we no longer use the rule name and it's not in the model anymore

**Review by will-larkin-nhs — CHANGES_REQUESTED** [review]
If the application won't run without this new function, there should be a docker implementation for it

File: application/CohortManager/src/Functions/AggregateDataServices/AddAggregationData/AddAggregationDataFunction.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using Common;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+using Model;
+
+namespace AddAggregationData;
```
Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/AggregateDataServices/AddAggregationData/AddAggregationDataFunction.cs:10]: Namespace should be `NHS.CohortManager.AggregationDataServices` as per convention, and be at the top of the file as it is a file-scoped namespace

File: application/CohortManager/src/Functions/AggregateDataServices/AddAggregationData/AddAggregationDataFunction.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+public class AddAggregationDataFunction
```
Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/AggregateDataServices/AddAggregationData/AddAggregationDataFunction.cs:12]: IMO classes, functions, etc should not include noise words like "Function", not a huge issue though

File: application/CohortManager/src/Functions/Shared/Common/Interfaces/ICreateAggregationData.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+using Model;
+
+namespace Common;
```
Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/Shared/Common/Interfaces/ICreateAggregationData.cs:3]: namespace should be before using statements

File: application/CohortManager/src/Functions/Shared/Data/Database/CreateAggregationData.cs
Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/Shared/Data/Database/CreateAggregationData.cs:1]: there seems to be no namespace in this file?

File: application/CohortManager/src/Functions/Shared/Model/AggregateParticipant.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
+using Model.Enums;
+
+namespace Model;
```
Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/Shared/Model/AggregateParticipant.cs:3]: namespace should be before using

File: tests/AddAggregationDataTests/AggregationDataTests/AddAggregatonTests.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
+namespace AggregationDataTests;
```
Comment (will-larkin-nhs) [inline tests/AddAggregationDataTests/AggregationDataTests/AddAggregatonTests.cs:1]: Namespace should be `NHS.CohortManager.Tests.AddAggregationDataTests`

File: tests/AddAggregationDataTests/AggregationDataTests/AddAggregatonTests.cs [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    [TestMethod]
+    public void InsertAggregationData_Success()
+    {
+        var createAggregationData = new CreateAggregationData(
+                _mockDBConnection.Object,
+                _databaseHelperMock.Object,
+                _loggerMock.Object,
+                _callFunction.Object
+            );
+        // Arrange
```
Comment (will-larkin-nhs) [inline tests/AddAggregationDataTests/AggregationDataTests/AddAggregatonTests.cs:62]: just have arrange here in the middle of the test. Should have arrange at the start, then act, then assert

**Review by will-larkin-nhs — CHANGES_REQUESTED** [review]

**Review by will-larkin-nhs — APPROVED** [review]

**Review by will-larkin-nhs — APPROVED** [review]

**Pull request merged**